lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -0,0 +1,146 @@
+require 'fastlane_core/ui/ui'
+require_relative '../actions/firebase_app_distribution_login'
+
+module Fastlane
+  module Client
+    module FirebaseAppDistributionApiClient # Class or module? I figured module sinze there is no state to track
```

Comment:
I think we should make this a class and make the methods below instance methods instead of class methods. At the moment there is technically not any configurable state, but I think it is likely that at some point it will be useful to have configurable state. For example, the caller could pass a flag to specify which login they would like to use as opposed to the client choosing the login method. Or the caller could pass a string specifying which environment to call i.e. prod, dev, test (this may be particularly useful for tests). Even though at the moment the `FirebaseAppDistributionAction` is the only caller, I think making this a class with instance methods (and maybe introducing configurable state to help with testing) is the better way to implement the api client.

Reply:
That sounds good. Do you think I should keep the methods that we currently have as class method or change some/all of them to instance methods?

Reply:
Instance methods. Also any methods that you don't plan to expose to callers should be private or protected.